endpoints-framework/src/main/java/com/google/api/server/spi/config/model/SchemaRepository.java

```diff
@@ -186,6 +206,42 @@ private void addSchemaToApi(ApiKey key, Schema schema) {
         addSchemaToApi(key, f.schemaReference().get());
       }
     }
+    Field mapValueSchema = schema.mapValueSchema();
+    if (mapValueSchema != null && mapValueSchema.schemaReference() != null)  {
+      addSchemaToApi(key, mapValueSchema.schemaReference().get());
+    }
+  }
+
+  private Optional<Schema> createMapSchema(
+      TypeToken type, Map<TypeToken<?>, Schema> typesForConfig, ApiConfig config) {
+    FieldType keyFieldType = FieldType.fromType(Types.getTypeParameter(type, 0));
+    boolean supportedKeyType = SUPPORTED_MAP_KEY_TYPES.contains(keyFieldType);
+    if (!supportedKeyType) {
+      String message = "Map field type '" + type + "' has a key type not serializable to String";
```

Comment:
It would make this slightly more compact too : 
```
boolean supportedKeyType = SUPPORTED_MAP_KEY_TYPES.contains(keyFieldType);
if (!supportedKeyType) {
```
=> 
```
if (!keyFieldType.isSerializableToString()) {
```

Reply:
See previous comment